sockeye/inference.py

```diff
@@ -603,13 +603,17 @@ def chunks(self, chunk_size: int) -> Generator['TranslatorInput', None, None]:
         :param chunk_size: The maximum size of a chunk.
         :return: A generator of TranslatorInputs, one for each chunk created.
         """
+        constraints = self.constraints
```

Comment:
I like the new fallback behavior for chunked inputs.
Could we clarify this code piece a little bit though?
Maybe something like this:
```python
for chunk_id, i in enumerate(range(0, len(self), chunk_size)):
  # Constrained decoding is not supported for chunked TranslatorInputs. Constraints are set only for the first chunk.
  if chunk_id == 0:
      constraints = self.constraints
  else:
      None
  ...
```

Another solution (to keep the code and the warning in the same place) could be to log the warning in the function here. Would that work?

Reply:
Sure, that's simpler, pushing now, thanks!